### PR TITLE
DOCSP-44706 - Type discriminator note

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-44706

### Staging Links

<li><a href=https://deploy-preview-342--mongodb-docs-csharp.netlify.app/fundamentals/serialization/polymorphic-objects>fundamentals/serialization/polymorphic-objects</a></li><li><a href=https://deploy-preview-342--mongodb-docs-csharp.netlify.app/upgrade/v3>upgrade/v3</a></li><li><a href=https://deploy-preview-342--mongodb-docs-csharp.netlify.app/whats-new>whats-new</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade/v3.txt [hunk 1]
```diff
@@ -203,4 +203,35 @@ Version 3.0 Breaking Changes
...
+  - Call the ``Aggregate().Match(item => item is T)`` method, as in the following example:
+
+    .. code-block:: csharp
+
+       collection.Aggregate().Match(item => item is T)
+  
+  To use either of the preceding methods on a collection, you can apply a hierarchical
+  discriminator to each class in the collection. See
+  the :ref:`Polymorphic Objects <csharp-discriminator-hierarchical>`
+  page to learn how to do so.
```
Comment (stephmarie17) [inline source/upgrade/v3.txt:227]: [n] I think it's just as clear to omit this:
```suggestion
  page to learn how.
```

**Review by stephmarie17 — APPROVED** [review]
LGTM with a tiny nit 🚢

**Review by rishitb-mongodb — APPROVED** [review]

File: source/fundamentals/serialization/polymorphic-objects.txt [hunk 2]
```diff
@@ -156,7 +156,19 @@ you must explicitly list each class you're looking for:
...
+       item.GetType() == typeof(Tiger));
+
+.. note:: OfType<T>() and the is Operator
+
+   When checking the type of a scalar discriminator, use the ``Where`` syntax shown in
+   the preceding code example. If you try to use the ``Aggregate().OfType<T>()`` method,
+   or if you pass an expression containing the ``is`` operator to the
+   ``Aggregate().Match()`` method, the driver throws an exception.
+
+.. _csharp-discriminator-hierarchical:
```
Comment (rstam) [inline source/fundamentals/serialization/polymorphic-objects.txt:171]: This limitation will go away when https://jira.mongodb.org/browse/CSHARP-5356 is merged.

**Review by rstam — APPROVED** [review]
LGTM

See the comment about CSHARP-5356

File: source/fundamentals/serialization/polymorphic-objects.txt (hunk 2 above)
Comment (mongoKart) [reply source/fundamentals/serialization/polymorphic-objects.txt:171]: We'll publish this now and can update in 3.2 or whenever the fix is implemented.

**Pull request merged**